src/Widgets/EntryPopover.vala

```diff
@@ -0,0 +1,134 @@
+/*
+* Copyright 2020 elementary, Inc. (https://elementary.io)
+*
+* This program is free software; you can redistribute it and/or
+* modify it under the terms of the GNU General Public
+* License as published by the Free Software Foundation; either
+* version 3 of the License, or (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+* General Public License for more details.
+*
+* You should have received a copy of the GNU General Public
+* License along with this program; if not, write to the
+* Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+* Boston, MA 02110-1301 USA
+*/
+
+public abstract class Tasks.EntryPopover<T> : Gtk.EventBox {
+    public Gtk.Image image { get; set; }
+    public Gtk.Popover popover { get; private set; }
+    public string? placeholder { get; set; }
+
+    public T value { get; set; }
+    private T value_on_popover_show { get; set; }
+    public signal void value_changed (T value);
+    public signal string? value_format (T value);
+
+    private static Gtk.CssProvider style_provider;
+    private Gtk.MenuButton popover_button;
+
+    class construct {
+        set_css_name ("entry-popover");
+    }
+
+    static construct {
+        style_provider = new Gtk.CssProvider ();
+        style_provider.load_from_resource ("io/elementary/tasks/EntryPopover.css");
+    }
+
+    construct {
+        events |= Gdk.EventMask.ENTER_NOTIFY_MASK
+            | Gdk.EventMask.LEAVE_NOTIFY_MASK;
+
+        popover = new Gtk.Popover (popover_button);
+
+        popover_button = new Gtk.MenuButton () {
+            always_show_image = true,
+            label = (placeholder != null && placeholder.length > 0 ? placeholder : _("Set Value")),
```

Comment:
I can't imagine a case where we ever want a button to say "Set Value". This should probably be a construction property instead and save translators the work for a string that will never be shown

Reply:
Sounds good!